### PR TITLE
Support external subtitles in external player

### PR DESCRIPTION
**Pull request by sunrisela**
…ayer, VLC, nPlayer.

**Changes**

modifies ExternalPlayerActivity.java and ExternalPlayerProfile.kt to support high-priority external subtitles for External players, e.g. MX Player, VLC, nPlayer.

**Issues**

### CODE REVIEW

File: app/src/main/java/org/jellyfin/androidtv/ui/playback/ExternalPlayerActivity.java [hunk 1]
```diff
@@ -402,6 +411,11 @@ protected void startExternalActivity(String path, String container) {
                 external.putExtra(API_MX_FILENAME, file.getName());
             }
         }
+
+        // updated by Joshua.Li
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/ui/playback/ExternalPlayerActivity.java:415]: Who is this Joshua?

File: app/src/main/java/org/jellyfin/androidtv/ui/playback/ExternalPlayerActivity.java [hunk 2]
```diff
@@ -417,4 +431,50 @@ protected void startExternalActivity(String path, String container) {
             handlePlayerError();
         }
     }
+
+    /**
+     * Adapt external subtitles for external players. (e.g., MX Player, MPV, VLC, nPlayer)
+     * External subtitles have higher priority than embedded subtitles.
+     *
+     * @param mediaStreamInfo
+     * @param playerIntent
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/ui/playback/ExternalPlayerActivity.java:440]: Just remove the params if they're not documented
```suggestion
```

File: app/src/main/java/org/jellyfin/androidtv/util/profile/ExternalPlayerProfile.kt [hunk 3]
```diff
@@ -18,54 +18,18 @@ class ExternalPlayerProfile : DeviceProfile() {
...
-				container = Codec.Container.MP3
-				audioCodec = Codec.Audio.MP3
 			}
 		)
 
+		transcodingProfiles = emptyArray()
+
+		containerProfiles = emptyArray()
+
+		codecProfiles = emptyArray()
```
Comment (nielsvanvelzen) [inline app/src/main/java/org/jellyfin/androidtv/util/profile/ExternalPlayerProfile.kt:31]: I'm quite sure those are empty arrays by default, so no need to reassign

**Review by sunrisela — COMMENTED** [review]
has changed.

File: app/src/main/java/org/jellyfin/androidtv/ui/playback/ExternalPlayerActivity.java (hunk 1 above)
Comment (nielsvanvelzen) [reply app/src/main/java/org/jellyfin/androidtv/ui/playback/ExternalPlayerActivity.java]: We don't credit authors in-code so this line can be removed

**Review by sunrisela — COMMENTED** [review]
has removed credit authors in-code

**Review by nielsvanvelzen — APPROVED** [review]

**Pull request merged**